mne/io/eeglab/eeglab.py

```diff
@@ -445,3 +471,45 @@ def __init__(self, input_fname, events=None, event_id=None, tmin=0,
             reject=reject, flat=flat, reject_tmin=reject_tmin,
             reject_tmax=reject_tmax, add_eeg_ref=False, verbose=verbose)
         logger.info('Ready.')
+
+
+def _events_from_eeglab_raw(fname, event_id=dict()):
+    """Create events array from EEGLAB structure by looking them up in the
+    event_id, trying to reduce them to their integer part otherwise, and
+    entirely dropping them (with a warning) if this is impossible"""
+    from scipy.io import loadmat
+
+    eeg = loadmat(fname, struct_as_record=False, squeeze_me=True)["EEG"]
+    types = [event.type for event in eeg.event]
+    latencies = [event.latency for event in eeg.event]
+
+    not_in_event_id = set([x for x in types if x not in event_id])
+    not_purely_numeric = set([x for x in not_in_event_id if not x.isdigit()])
+    no_numbers = set([x for x in not_purely_numeric
+                      if not any([d.isdigit() for d in x])])
+    have_integers = set([x for x in not_purely_numeric
+                         if x not in no_numbers])
+    if len(not_purely_numeric) > 0:
+        basewarn = "Events like the following will be "
+        n_no_numbers, n_have_integers = len(no_numbers), len(have_integers)
+        if n_no_numbers > 0:
+            nonumwarm = "dropped entirely: {}, {} in total"
+            warnings.warn(basewarn + nonumwarm.format(list(no_numbers)[:5],
+                                                      n_no_numbers))
+        if n_have_integers > 0:
+            intwarn = "reduced to their integer part: {}, {} in total"
+            warnings.warn(basewarn + intwarn.format(list(have_integers)[:5],
+                                                    n_have_integers))
+        warnings.warn("Use the `event_id` keyword to "
+                      "include such events manually.")
+
+    events = list()
+    for t, latency in zip(types, latencies):
+        try:
+            event_code = event_id.get(t, int("".join([x for x in t
```

Comment:
I don't get what you mean. 

The event readout part basically works. An empty `event_id=dict()` is how it's done in the BV reader. What this one here does, as-is, is 1. look up the key in the event_id, 2. if nothing is found in the event_id (for example, if `event_id` is empty), retrieve the integer part instead, 3. warn if any keys will be empty, and warn if non-integer parts will be dropped anywhere. Thus, with this code, event reading works, if you set `preload=True` - see the ERP in the 2nd post.

The problem is I don't get how to integrate it into incremental reading (`preload=False`).